WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -454,6 +440,31 @@ private async Task HandleCoinJoinFinalizationAsync(CoinJoinTracker finishedCoinJ
 			wallet.LogError($"{nameof(CoinJoinClient)} failed with exception: '{e}'");
 		}
 
+		NotifyCoinJoinCompletion(finishedCoinJoin);
```

Comment:
You should use StopWhenAllMixed somewhere here.

Reply:
Could you please explain to me again, what is the `StopWhenAllMixed` responsible for?

AFAIK it only indicates that the `AutoCoinJoin` is on/off.

So, if all of our coins are private then =>  this check will throw

```cs
if (await walletToStart.IsWalletPrivateAsync().ConfigureAwait(false) && startCommand.StopWhenAllMixed)
{
       walletToStart.LogTrace("All mixed!");

	throw new CoinJoinClientException(CoinjoinError.AllCoinsPrivate);
}
```

and it will be handled here (we will show the correct musicBox message (Hurray! ...) and keep retrying)

```cs
else if (cjClientException is not null)
{
	// Keep trying, so CJ start automatically when the wallet becomes mixable again.
	ScheduleRestartAutomatically(wallet, trackedAutoStarts, finishedCoinJoin.StopWhenAllMixed, finishedCoinJoin.OverridePlebStop, cancellationToken);
	NotifyCoinJoinStartError(wallet, cjClientException.CoinjoinError);
}
```

Where and why should we use  `StopWhenAllMixed`?

Reply:
So this is the exact place to investigate. if StopWhenAllMixed is true then we should not restart and keep retrying.

Reply:
Please check the new commit [ccd5a18](https://github.com/zkSNACKs/WalletWasabi/pull/10527/commits/ccd5a18773a950025af4884d891375b75de56090)

I think I get the logic here.

With that commit, if the wallet is private and autocoinjoin is off, then we don't restart the CJ and we don't keep on trying.
With AutoCoinjoin ON, we keep on restarting and we keep on trying to do CJs, so when new coins arrive to the wallet, we will start CJing with them.

Reply:
Perfect.